ESM module: "Could not find a declaration file for module 'typeorm'" (0.3.6)

## Issue Description

A minimal ESM module with only TypeORM does not compile.

### Expected Behavior

The project should compile without error.

### Actual Behavior

```
$ yarn build
yarn run v1.22.18
warning package.json: No license field
$ tsc
src/index.ts:1:17 - error TS7016: Could not find a declaration file for module 'typeorm'. '/home/arsleust/perso/tmp/esm-module-ts-typeorm/node_modules/typeorm/index.mjs' implicitly has an 'any' type.
  Try `npm i --save-dev @types/typeorm` if it exists or add a new declaration (.d.ts) file containing `declare module 'typeorm';`

1 import orm from "typeorm"
                  ~~~~~~~~~


Found 1 error in src/index.ts:1

error Command failed with exit code 2.
```

```
// include the output in code tags like these!
```


### Steps to Reproduce

`package.json`
```json
{
  "name": "mypackage",
  "type": "module",
  "scripts": {
    "build": "tsc"
  },
  "devDependencies": {
    "typescript": "^4.8.0-dev.20220605"
  },
  "dependencies": {
    "typeorm": "^0.3.6"
  }
}
```

`tsconfig.json`
```
{
  "compilerOptions": {
    "target": "es2018",
    "module": "NodeNext",
    "moduleResolution": "NodeNext",
    "esModuleInterop": true,
    "strict": true,
    "skipLibCheck": true
  }
}
```

`src/index.ts`
```ts
import orm from "typeorm"

```

Run `yarn build`

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Node.js version     | 16.14.2  | <!-- run `node -v` to obtain this -->
| Typescript version  | 4.8.0-dev.20220605  | <!-- run `npm list typescript` to obtain this -->
| TypeORM version     | 0.3.6  | <!-- run `npm list typeorm` to obtain this -->


### Are you willing to resolve this issue by submitting a Pull Request?

<!-- Put "✅" (:white_check_mark:) to one of these options, left "✖️" (:heavy_multiplication_x:) others: -->

 - ✖️ Yes, I have the time, and I know how to start.
 - ✅ Yes, I have the time, but I don't know how to start. I would need guidance.
 - ✖️ No, I don’t have the time, but I can support (using donations) development.
 - ✖️ No, I don’t have the time and I’m okay to wait for the community / maintainers to resolve this issue.


<!--
  Remember that first-time contributors are welcome! 🙌
  👋 Have a great day and thank you for the bug report!
-->

--- a/package.json
+++ b/package.json
@@ -17,7 +17,8 @@
       "types": "./index.d.ts",
       "node": {
         "import": "./index.mjs",
-        "require": "./index.js"
+        "require": "./index.js",
+        "types": "./index.d.ts"
       },
       "browser": {
         "require": "./index.js",

